staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go

```diff
@@ -194,6 +225,51 @@ func TestWatchErrorIncorrectConfiguration(t *testing.T) {
 	}
 }
 
+func TestTooLargeResourceVersionErrorForWatchList(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.WatchList, true)()
+	origCtx, store, _ := testSetup(t)
+	ctx, cancel := context.WithCancel(origCtx)
+	defer cancel()
+	requestOpts := storage.ListOptions{
+		SendInitialEvents: ptr.To(true),
+		Recursive:         true,
+		Predicate: storage.SelectionPredicate{
+			Field:               fields.Everything(),
```

Comment:
nit: I think you don't need to set them explicitly if you want everything

Reply:
I can change to `fields.Noting()` if that helps :)
I need to init the memory.

Reply:
I thought we're handling nil - if not, let's leave as is.